test/model/src/core/kv_table.cpp

```diff
@@ -356,6 +376,36 @@ kv_table::verify_cursor()
     return std::move(kv_table_verify_cursor(_data));
 }
 
+/*
+ * kv_table::fill_missing_column_fix_recnos --
+ *     Fill in missing recnos for FLCS to ensure that key ranges are contiguous.
+ */
+void
+kv_table::fill_missing_column_fix_recnos_nolock(const data_value &key)
+{
+    if (_config.type != kv_table_type::column_fix)
+        return;
+
+    if (!std::holds_alternative<uint64_t>(key))
+        throw model_exception("The key is not compatible with a column store: Not a recno.");
+    uint64_t recno = std::get<uint64_t>(key);
+
+    if (_data.empty()) {
+        for (uint64_t i = 1; i <= recno; i++)
+            _data[data_value(i)].add_update(
+              std::make_shared<kv_update>(ZERO, k_timestamp_none, true /* implicit */), false,
+              false);
+    } else {
+        if (!std::holds_alternative<uint64_t>(_data.begin()->first))
+            throw model_exception("Invalid keys in a column store: Not a recno.");
+        uint64_t last = std::get<uint64_t>(_data.rbegin()->first);
+        for (uint64_t i = last + 1; i <= recno; i++)
+            _data[data_value(i)].add_update(
+              std::make_shared<kv_update>(ZERO, k_timestamp_none, true /* implicit */), false,
+              false);
+    }
```

Comment:
Feel free to ignore, I thought we could avoid the "same" loop twice:
```suggestion
    uint64_t last = 0;
    if (!_data.empty()) {
        if (!std::holds_alternative<uint64_t>(_data.begin()->first))
            throw model_exception("Invalid keys in a column store: Not a recno.");
        last = std::get<uint64_t>(_data.rbegin()->first);
    }
    for (uint64_t i = last + 1; i <= recno; i++)
        _data[data_value(i)].add_update(
            std::make_shared<kv_update>(ZERO, k_timestamp_none, true /* implicit */), false,
            false);
```

Reply:
Good idea—let me take that in. Thanks!